src/main/java/org/folio/innreach/controller/ContributionCriteriaController.java

```diff
@@ -0,0 +1,133 @@
+  public ResponseEntity<Void> deleteCriteria(@PathVariable UUID centralServerId) {
+    criteriaConfigurationService.delete(centralServerId);
+    return ResponseEntity.noContent().build();
+  }
+
+  @Override
+  @GetMapping("/{centralServerId}/contribution-criteria")
+  public ResponseEntity<ContributionCriteriaDTO> getCriteriaById(@PathVariable UUID centralServerId) {
+    return ResponseEntity.ok(
+      toContributionCriteriaDTO(
+        criteriaConfigurationService.get(centralServerId)
+      )
+    );
+  }
+
+  @Override
+  @PutMapping("/{centralServerId}/contribution-criteria")
+  public ResponseEntity<ContributionCriteriaDTO> updateCriteria(@PathVariable UUID centralServerId,
+                                                            @Valid ContributionCriteriaDTO contributionCriteriaDTO) {
+    return ResponseEntity.ok(toContributionCriteriaDTO(
+      criteriaConfigurationService.update(toContributionCriteriaConfigurationDTO(contributionCriteriaDTO))
+    ));
+  }
+
+  public ContributionCriteriaDTO toContributionCriteriaDTO (ContributionCriteriaConfigurationDTO contributionCriteriaConfigurationDTO) {
+    var result = new ContributionCriteriaDTO();
+    result.setCentralServerId(contributionCriteriaConfigurationDTO.getCentralServerId());
+    var metadata = new Metadata();
+    metadata.setCreatedByUserId(contributionCriteriaConfigurationDTO.getCreatedBy());
+    metadata.setUpdatedByUserId(contributionCriteriaConfigurationDTO.getLastModifiedBy());
+    metadata.setCreatedDate(dateMapper.offsetDateTimeAsDate(contributionCriteriaConfigurationDTO.getCreatedDate()));
+    metadata.setUpdatedDate(dateMapper.offsetDateTimeAsDate(contributionCriteriaConfigurationDTO.getLastModifiedDate()));
+    result.setMetadata(metadata);
+    contributionCriteriaConfigurationDTO.getExcludedLocations().stream().forEach(excludedLocationDTO -> result.addLocationIdsItem(excludedLocationDTO.getExcludedLocationId()));
+    contributionCriteriaConfigurationDTO.getStatisticalCodeBehaviors().stream().forEach(statisticalCodeBehaviorDTO -> {
+      switch (statisticalCodeBehaviorDTO.getContributionBehavior()) {
+        case doNotContribute:
+          result.setDoNotContributeId(statisticalCodeBehaviorDTO.getStatisticalCodeId());
+          break;
+        case contributeButSuppress:
+          result.setContributeButSuppressId(statisticalCodeBehaviorDTO.getStatisticalCodeId());
+          break;
+        case contributeAsSystemOwned:
+          result.setContributeAsSystemOwnedId(statisticalCodeBehaviorDTO.getStatisticalCodeId());
+          break;
+        default:
+          throw new IllegalArgumentException("Behavior of Contribution Criteria: "+statisticalCodeBehaviorDTO.getContributionBehavior().toString() +" can not be handled!");
+      }
+    });
+    return  result;
+  }
+
+  public ContributionCriteriaConfigurationDTO toContributionCriteriaConfigurationDTO(ContributionCriteriaDTO contributionCriteriaDTO) {
```

Comment:
Why are you doing the DTO conversion if the service is already returning a DTO class?
We usually do this transformation not in controllers, but in services.
Why do you need a separate method for this? You should use MapStruct converters.